zerver/tests/test_previewable.py

```diff
@@ -0,0 +1,212 @@
+import json
+from typing import Dict, Tuple
+
+import requests
+import responses
+from django.test import override_settings
+
+from zerver.lib.test_classes import ZulipTestCase
+
+
+class TestPreviewable(ZulipTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.login("hamlet")
+
+    @override_settings(GITHUB_API_AUTH_TOKEN="3608bca1e44ea6c4d268eb6db02260269892c0")
+    @responses.activate()
+    def test_github_auth(self) -> None:
+        def request_callback(
+            request: requests.PreparedRequest,
+        ) -> Tuple[int, Dict[str, str], str]:
+            if request.headers["Authorization"] == "token 3608bca1e44ea6c4d268eb6db02260269892c0":
+                return (200, {}, json.dumps({}))
+            return (403, {}, json.dumps({}))  # nocoverage
+
+        responses.add_callback(
+            responses.GET,
+            "https://api.github.com/repos/zulip/zulip/issues/19710",
+            callback=request_callback,
+            content_type="application/json",
+        )
+        result = self.client_post(
+            "/json/previewable", {"url": "https://github.com/zulip/zulip/pull/19710"}
+        )
+        self.assertEqual(result.status_code, 200)
+
+    @responses.activate()
+    def test_github_without_auth(self) -> None:
+        def request_callback(request: requests.PreparedRequest) -> Tuple[int, Dict[str, str], str]:
+            if (
+                request.headers.get("Authorization", None)
+                == "token 3608bca1e44ea6c4d268eb6db02260269892c0"
+            ):
+                return (200, {}, json.dumps({}))  # nocoverage
+            return (403, {}, json.dumps({}))
```

Comment:
You have this code twice, slightly differently -- and in neither case do you actually _need_ the `request_callback`, since in each case you only exercise one side of the `if` statement.

You should pull this helper outside of your two test methods, so it gets exercised.